app/src/main/kotlin/org/kotlinlsp/common/Log.kt

```diff
@@ -29,21 +33,61 @@ private enum class LogLevel(level: Int) {
 private val logLevel = LogLevel.Debug
 private const val profileEnabled = true
 
-private lateinit var logFile: File
+private lateinit var logger: LSPLogger
 private val profileInfo = mutableMapOf<String, Pair<Int, Duration>>()
 
-fun setupLogger(path: String) {
-    val loggerPath = "$path/log.txt"
-    logFile = File(loggerPath)
-    if (logFile.exists()) {
-        logFile.delete()
+private class LSPLogger {
+    var client: LanguageClient? = null
+    val messageQueue = ConcurrentLinkedQueue<MessageParams>()
+
+    fun log(level: LogLevel, message: String) {
+        if (level < logLevel) return
+
+        val type = when (level) {
```

Comment:
Maybe we can move this to a `LogLevel.asMessageType` extension function

Reply:
Done